dev/puppet-server.conf.sample

```diff
@@ -65,6 +65,18 @@ http-client: {
     #                TLS_RSA_WITH_AES_256_CBC_SHA,
     #                TLS_RSA_WITH_AES_128_CBC_SHA256,
     #                TLS_RSA_WITH_AES_128_CBC_SHA]
+
+    # The amount of time, in milliseconds, that an outbound HTTP connection
+    # will wait for no data to be available before closing the socket. If not
+    # defined, defaults to 20 minutes. If 0, the timeout is infinite and if
+    # negative, the value is undefined (system default).
+    #socket-timeout: 1200000
+
+    # The amount of time, in milliseconds, that an output HTTP connection will
```

Comment:
You should see some of my comments from the Puppet codebase.  "Let me see if I understand my grip on reality..." comes to mind.